### DISCUSSION TITLE
Naming of patches

### DISCUSSION THREAD

**Discussion by Technikte** [ReVanced/revanced-discussions#471] [Question]
The problem I see is that when e.g. patches for subscription unlock exist for many apps later and all are called "subscription-unlock". What is the best or recommended way to deal with this?

Each time in the patch name to include the app name like "general-ads-reddit" or in some other way.
Next question if this matters at all if the revanced manager or other 3rd party tools show only matching patches after app selection.

This issue comes mainly to the surface later times when many pachtes exist.

**Comment by oSumAtrIX** [comment]
This is indeed a problem i have encountered as well, i think in the manager it actually does not cause any problems, theres just ambiguity in the cli. Even there the compatibility annotation actually would filter out the patches for other apps. If you have the same patch name within the same package that could cause problems, do you have a suggestion how to deal with it? My beat bet would be contribution guidelines.

**Comment by Technikte** [comment]
I dont have any really good suggestion at the moment, for the time we could suffix all patch names with the app names like the reddit patches. At some point there must be some contribution guidelines otherwise it could get pretty messy with all patches.

I wonder what the other dev would suggest, so I will ping them here. 
@Sculas @TheJeterLP @Canny1913 @CnC-Robert @danthe1st

**Reply by oSumAtrIX** [reply]
Suffixing the package names is not solving conflicts within a package itself. So if theres two patches with the same name targeting one app, it will conflict. If they target different apps, but patches would technically be told the cli to include, but the package name check would prevent one of them to be added to the patcher.

**Reply by Technikte** [reply]
For sure this wont fix this, I wonder how often this will occur 2 patches with same for same app..maybe only if there are different ways to achive a goal.

**Reply by oSumAtrIX** [reply]
In that case the problem currently is not a problem.

**Reply by danthe1st** [reply]
One possibility is to introduce something similar to group ids/namespaces/packages.

In addition to the name, patches get a group id (or namespace). The group id should be unique to the developer. The combination between group id (package to patch(?)) and patch name should then be unique if each patch developer makes sure that they don't use the same name multiple times.

As long as there are no patches with the same name, displaying the patch name (and selecting patches by name) should be sufficient. If multiple patches with the same name exist, they need to be fully qualified with the group id and patch name.

For choosing a group id, one option would be reverse domain name notation which is already common in the JVM ecosystem.
